src/Compilers/VisualBasic/Test/Semantic/IOperation/IOperationTests_ArrayCreationAndInitializer.vb

```diff
@@ -0,0 +1,435 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Semantics
+Imports Microsoft.CodeAnalysis.Test.Utilities
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Roslyn.Test.Utilities
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics
+
+    Partial Public Class IOperationTests
+        Inherits SemanticModelTestBase
+
+        <Fact(Skip:="https://github.com/dotnet/roslyn/issues/18150"), WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub SimpleArrayCreation_PrimitiveType()
+            Dim source = <![CDATA[
+Class C
+    Public Sub F()
+        Dim a = New String(0) {}'BIND:"New String(0) {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: System.String) (OperationKind.ArrayCreationExpression, Type: System.String())
+  ILiteralExpression (Text: 0) (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub SimpleArrayCreation_UserDefinedType()
+            Dim source = <![CDATA[
+Class M
+End Class
+
+Class C
+    Public Sub F()
+        Dim a = New M() {}'BIND:"New M() {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M())
+  ILiteralExpression (OperationKind.LiteralExpression, Type: System.Int32, Constant: 0)
+  IArrayInitializer (OperationKind.ArrayInitializer)
+]]>.Value
+
+            VerifyOperationTreeForTest(Of ArrayCreationExpressionSyntax)(source, expectedOperationTree)
+        End Sub
+
+        <Fact, WorkItem(17596, "https://github.com/dotnet/roslyn/issues/17596")>
+        Public Sub SimpleArrayCreation_ConstantDimension()
+            Dim source = <![CDATA[
+Class M
+End Class
+
+Class C
+    Public Sub F()
+        Const dimension As Integer = 1
+        Dim a = New M(dimension) {}'BIND:"New M(dimension) {}"
+    End Sub
+End Class
+    ]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IArrayCreationExpression (Dimension sizes: 1, Element Type: M) (OperationKind.ArrayCreationExpression, Type: M())
+  IBinaryOperatorExpression (BinaryOperationKind.IntegerAdd) (OperationKind.BinaryOperatorExpression, Type: System.Int32, Constant: 2)
```

Comment:
That is how the language works, if you specified 1, the size of the allocated array will be two (one is added). I believe this is for compatibility with VB6, which had one-based arrays. So, it was common to have VB6 code to allocate an array of size N and then index into the array using N as index.